python/ray/workflow/workflow_executor.py

```diff
@@ -309,6 +318,7 @@ async def _post_process_ready_task(
         output_ref: WorkflowRef,
     ) -> None:
         state = self._state
+        state.task_retries.pop(task_id, None)
```

Comment:
Ok for now, but I feel this is generally easy to make mistake by just forgetting the cleanup here.